File: source/downloads/code/LeetCode/Python/Sort-Colors.py
class Solution(object):
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        if not nums or len(nums) == 0:
            return

        indexR = 0
        indexB = len(nums)-1

        i = 0
        while i <= indexB:
            if nums[i] == 0:
                nums[indexR],nums[i] = nums[i],nums[indexR]
                indexR += 1
                i += 1
            elif nums[i] == 2:
                nums[indexB],nums[i] = nums[i],nums[indexB]
                indexB -= 1
            else:
                i += 1

File: downloads/code/LeetCode/Java/Sort-Colors.java
public class Solution {
    public void sortColors(int[] nums) {
        if(nums == null || nums.length==0)
            return;
        int red = 0;
        int blue = nums.length-1;
        
        int i=0;
        while(i<=blue){
            if(nums[i]==0){
                swap(nums,red,i);
                red++;
                i++;
            }
            else if (nums[i]==2){
                swap(nums,blue,i);
                blue--;
            }
            else
                i++;
        }
        return ; 

    }

    public void swap(int[] nums, int a, int b){
        int tmp = nums[a];
        nums[a] = nums[b];
        nums[b] = tmp;
    }
}
